mnd/mnd/views/carer_registration_view.py

```diff
@@ -0,0 +1,204 @@
+
+        return render(
+            self.request,
+            "registration/carer_enroll_token.html",
+            context={"email": self.primary_carer.email}
+        )
+
+
+class PatientCarerRegistrationView(LoginRequiredMixin, View):
+
+    def primary_carer_str(self, primary_carer):
+        if primary_carer:
+            return f"{primary_carer.first_name} {primary_carer.last_name} ({primary_carer.email})"
+        else:
+            return _("No primary carer set !")
+
+    def render_form(self, request, form, primary_carer, flags):
+        return render(
+            request,
+            "registration/carer_enroll.html",
+            context={
+                "form": form,
+                "carer": self.primary_carer_str(primary_carer),
+                "flags": flags,
+            }
+        )
+
+    def get(self, request):
+        if not request.user.is_patient:
+            return HttpResponseForbidden()
+        patient = request.user.user_object.first()
+        primary_carer = PrimaryCarer.objects.filter(patients__id=patient.id).first()
+        flags = RegistrationFlags()
+        flags.update(primary_carer, patient)
+        if not flags.can_register:
+            form = PrimaryCarerForm(data=primary_carer.__dict__) if primary_carer else PrimaryCarerForm(data={})
+            if not (flags.can_deactivate or flags.can_activate):
+                if primary_carer:
+                    form.add_error(None, ValidationError(_("A token was already generated for this carer !")))
+                else:
+                    form.add_error(None, ValidationError(_("Primary carer need to be set in order to register it !")))
+        else:
+            form = PrimaryCarerForm(instance=primary_carer)
+        return self.render_form(request, form, primary_carer, flags)
+
+    def post(self, request):
+        if not request.user.is_patient:
+            return HttpResponseForbidden()
+        patient = request.user.user_object.first()
+        primary_carer = PrimaryCarer.objects.filter(patients__id=patient.id).first()
+        operations = CarerOperations(request, primary_carer, patient)
+        if CarerRegistration.objects.has_pending_registration(primary_carer):
+            return HttpResponseForbidden()
+
+        if CarerRegistration.objects.has_registered_carer(primary_carer, patient):
+            return operations.deactivate_carer()
+
+        if CarerRegistration.objects.has_deactivated_carer(primary_carer, patient):
+            return operations.activate_carer()
+
+        # resending email invite for expired carers
+        # is the same as registering a carer
+        return operations.register_carer()
+
+
+class CarerRegistrationView(RegistrationView):
+
+    def get(self, request, registry_code):
+        request.session['REGISTRATION_CLASS'] = 'mnd.registry.groups.registration.mnd_registration.MNDCarerRegistration'
+        request.session['REGISTRATION_FORM'] = 'mnd.forms.mnd_registration_form.MNDCarerRegistrationForm'
```

Comment:
It would be nice if we could avoid setting these classes in the session.
Why can't we just subclass `RdrfRegistrationView` and set these classes on that subclass and use the subclass in `urls.py`?

Reply:
It was mainly because here we have both patient and carer registration and I didn't want to write an extra view just for handling a different registration and form class. I'll look into this some more maybe I can find a better alternative.

Reply:
Updated the code to use subclassing.